test/e2e/nf3lib/nf3.test.mjs

```diff
@@ -0,0 +1,39 @@
+/* eslint no-empty: "off" */
+
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import chaiAsPromised from 'chai-as-promised';
+import config from 'config';
+import { fail } from 'assert';
+import Nf3 from '../../../cli/lib/nf3.mjs';
+
+chai.use(chaiHttp);
+chai.use(chaiAsPromised);
+
+const environment = config.ENVIRONMENTS[process.env.ENVIRONMENT] || config.ENVIRONMENTS.localhost;
+
+const { mnemonics, signingKeys } = config.TEST_OPTIONS;
+
+describe('Nf3 Test', () => {
+  describe('Client Authenticated calls tests', () => {
+    it('Should fail since no authentication key is set', async () => {
+      const nf3User = new Nf3(signingKeys.user1, environment);
+      try {
+        await nf3User.init(mnemonics.user1);
+        fail();
```

Comment:
I think we should add some explanation at the top of this file. Because I don't think this test suite makes much sense outside the Github Action test `client-authentication-test` context. So one could simply bump into it and be like...

![image](https://user-images.githubusercontent.com/7704645/214917550-b50dedf0-6200-4c1f-81ea-52999a22e99e.png)

If you run the suite via `npm run test-e2e-nf3lib`, this particular test fails but not because of the described reason, because `fail()` will raise an `Assertion` error.

Reply:
Just to be clear
Test is `Should fail since no authentication key is set`
This passes, ie it fails, but due to `fail()` raising exception :see_no_evil:

Reply:
Yes, I agree with @imagobea. This is failing because of fail(). We need better explanation of this test.

Reply:
Yes, you're right. The tests are not well designed and miss some sort of documentation. Working on it! ;)

Reply:
Yes, you're right. The tests are not well designed and miss some sort of documentation. Working on this! ;)

Reply:
Yes, you're right. The tests are not well designed and miss some sort of documentation. Working on this! ;)

Reply:
I don't know what's going on on GitHub these days... Comments etc appear a gazillion times :rofl: